javers-core/src/main/java/org/javers/snapshots-compiler/JaversSnapshotsCompiler.java

```
@@ -0,0 +1,216 @@
+
+        Object instance = clientsClass.newInstance();
+        idToEntityMap.put(((InstanceId)snap.getGlobalId()).getCdoId(), instance);
+        //for (Property property : snap.getState().getProperties()) {
+        for (String propertyName : snap.getState().getProperties()) {
+            Object propertyValue = snap.getPropertyValue(propertyName);
+            Object propertyValueToSet = propertyValue;
+            if (propertyValue != null) {
+                try {
+                    Method setter = getSetterForProperty(clientsClass, propertyName);
+                    Class<?> targetType = setter.getParameterTypes()[0];
+                    if (targetType.isAssignableFrom(java.util.List.class)
+                            || targetType.isAssignableFrom(java.util.Set.class)) {
+                        Collection collection = (Collection) propertyValue;
+                        Set toBeDeleted = new HashSet<>();
+                        for (Object element : collection.toArray()){
+                            if (element instanceof InstanceId) {
+                                InstanceId instanceId = (InstanceId) element;
+
+                                Object targetObject = compileEntityForCommitId(
+                                        instanceId, snap.getCommitId());
+                                collection.add(targetObject);
+                                toBeDeleted.add(element);
+                            }
+                        }
+                        for (Object element : toBeDeleted) {
+                            collection.remove(element);
+                        }
+                    } else if (propertyValue instanceof InstanceId) {
+                        propertyValueToSet = compileEntityForCommitId(
+                                (InstanceId) propertyValue, snap.getCommitId());
+                    }
+                    setter.invoke(instance, propertyValueToSet);
+                } catch (Exception exc) {
```

swallowing all Exceptions is generally an antipatten, wrap it in JaversRuntimeException and rethrow